.github/workflows/health_checks.yml

```diff
@@ -269,10 +269,31 @@ jobs:
         run: |
           mkdir api-validation-projects
           npx tsx scripts/check_api_changes.ts base-branch-content api-validation-projects
+  generate_changeset:
+    if: github.event_name == 'pull_request' && github.event.pull_request.user.login == 'dependabot[bot]'
+    runs-on: ubuntu-latest
+    needs:
+      - install
+      - resolve_inputs
+    steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # version 4.1.4
+      - uses: ./.github/actions/setup_node
+        with:
+          node-version: 18
+      - uses: ./.github/actions/restore_install_cache
+        with:
+          node-version: 18
+          cdk-version: ${{ needs.resolve_inputs.outputs.cdk_version }}
+      - name: Generate changesets for packages with version updates
+        run: npx tsx scripts/dependabot_handle_version_update.ts "$BASE_SHA" "$HEAD_SHA"
+        env:
+          BASE_SHA: ${{ github.event.pull_request.base.sha }}
+          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
   do_include_e2e:
     needs:
       - install
       - resolve_inputs
+      - generate_changeset
```

Comment:
@rtpascual 

I think we can just remove this dependency here to fix the workflow on main.

Because:
First iteration of dependabot PR will add label AND push change to PR branch.

PR branch will trigger second independent run of workflow which will have label already added.